crates/bevy_app/src/plugin.rs

```diff
@@ -19,7 +52,7 @@ use std::any::Any;
 /// When adding a plugin to an [`App`]:
 /// * the app calls [`Plugin::build`] immediately, and register the plugin
 /// * once the app started, it will wait for all registered [`Plugin::ready`] to return `true`
-/// * it will then call all registered [`Plugin::finish`]
+/// * it will then call all registered [`Plugin::finalize`]
```

Comment:
Don't we need to call out the other life cycle steps added in this PR in these docs?

Reply:
👍 I'll fix all pending docs comments after the general code review, if that's ok (but open to do otherwise if you feel this is better)